### PR TITLE
DM-41609: preliminary kubernetes manifests for `cm-service` at USDF

### PR DESCRIPTION
**Pull request by fritzm**
In order to move on with the next round of integration testing, we need preliminary deployments of postgres and redis within the `usdf-cm-dev` vcluster at USDF.  These can serve as the shared backend to support developer `cm-service` FastAPI service instances running in debug mode on Rubin developer nodes.

This PR adds manifests for use with kustomize (e.g. `kubectl apply -k ...`).  The deployed postgres and redis instances are password secured (passwords stored in SLAC vault) and passwords are consumed via kubernetes secrets objects.  No secrets are committed in the manifests or makefile.

A `make` target is provided to fetch secrets from vault to a .gitignored, mode 700 directory in the local filesystem, where they can be picked up by kustomize and rendered into kubernetes secrets objects at `apply -k` time via kustomize's `secretGenerator`.  This target requires the developer to have pre-authorized via the vault CLI.  This pattern is based on prior usage in various slaclab repos (e.g. https://github.com/slaclab/slac-k8s-app-template).

(Presumably something better could be arranged that doesn't put unencrypted secrets at rest in the developer's file system, but it is unclear if USDF has yet favored any of the several available vault/k8s direct integrations, and if so, which?  Hoping this is a reasonable enough start for now.)

### CODE REVIEW

**Comment by fritzm** [comment]
Intending to still add `manifests/README.md` to this PR, with description of file layout, basic theory of operation, and `vault`\`kubectl` playbook for deploy/delete.

**Review by eacharles — APPROVED** [review]
This code, while quite awesome, is not as jaw-droppingly awe inspiring as the last big commit.

File: manifests/base/redis-lb.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: cm-redis-lb
+spec:
+  type: LoadBalancer
```
Comment (ktlim) [inline manifests/base/redis-lb.yaml:6]: Do you need to access this from elsewhere?  Maybe a ClusterIP is sufficient?

File: Makefile [hunk 2]
```diff
@@ -21,35 +23,72 @@ init:
...
+.PHONY: run-usdf-dev
+run-usdf-dev: CM_DATABASE_HOST=$(shell kubectl --cluster=usdf-cm-dev -n cm-service get svc/cm-pg-lb -o jsonpath='{..ingress[0].ip}')
+run-usdf-dev: CM_ARQ_REDIS_HOST=$(shell kubectl --cluster=usdf-cm-dev -n cm-service get svc/cm-redis-lb -o jsonpath='{..ingress[0].ip}')
+run-usdf-dev: export CM_DATABASE_URL=postgresql://cm-service@${CM_DATABASE_HOST}:5432/cm-service
+run-usdf-dev: export CM_ARQ_REDIS_URL=redis://${CM_ARQ_REDIS_HOST}:6379/1
+run-usdf-dev: export CM_DATABASE_PASSWORD=$(shell kubectl --cluster=usdf-cm-dev -n cm-service get secret/cm-pg-app -o jsonpath='{.data.password}' | base64 --decode)
+run-usdf-dev: export CM_ARQ_REDIS_PASSWORD=$(shell kubectl --cluster=usdf-cm-dev -n cm-service get secret/cm-redis-app -o jsonpath='{.data.password}' | base64 --decode)
+run-usdf-dev: export CM_DATABASE_ECHO=true
+run-usdf-dev:
+	cm-service init
```
Comment (ktlim) [inline Makefile:93]: So the `cm-service` itself is not running as a deployment?

**Review by ktlim — APPROVED** [review]

File: manifests/base/redis-lb.yaml (hunk 1 above)
Comment (fritzm) [reply manifests/base/redis-lb.yaml:6]: The deployment on this PR is to facilitate running the FastAPI part of the service in debug mode "in-hand" on rubin dev nodes, with just redis and postgres in k8s.  Safir needs access to redis, thus the lb.

File: Makefile (hunk 2 above)
Comment (fritzm) [reply Makefile:93]: Correct -- the deployment on this PR is to facilitate running the FastAPI part of the service in debug mode "in-hand" on rubin dev nodes, with just redis and postgres in k8s.

File: manifests/usdf-cm-dev/Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+.secrets/pg: .secrets
+	mkdir -p -m 700 .secrets/pg
+
+.secrets/redis: .secrets
+	mkdir -p -m 700 .secrets/redis
+
+.PHONY: fetch-vault-secrets
+fetch-vault-secrets: export VAULT_ADDR=https://vault.slac.stanford.edu
+fetch-vault-secrets: .secrets/pg .secrets/redis
+	vault kv get -mount=secret -field=username rubin/usdf-cm-dev/postgres > .secrets/pg/username
```
[the remark this answers is not in the document]
Comment (fritzm) [reply manifests/usdf-cm-dev/Makefile:15]: Will do.

**Pull request merged**